### ISSUE TITLE
Fix S101 FP: IEnumerableExtensionsTests should not be reported

### ISSUE THREAD

**Issue by NelsonPRSousa** [SonarSource/sonar-dotnet#3268]
### Description

I've created a new class called IEnumerableExtensionsTests so I can add unit tests to my static class IEnumerableExtensions, containing some extensions methods to the IEnumerable interface.

Weirdly, my static class doesn't trigger this warning.

IEnumerableExtensionsTests: Trigger the warning (public class)
IEnumerableExtensions: Does not trigger the warning (internal static class)
Rule Id: S101

### Repro steps

```csharp
namespace Mynamespace.Extensions
{
    using System.Collections.Generic;
    using System.Linq;

    internal static class IEnumerableExtensions
    {
        public static bool SafeAny<T>(this IEnumerable<T> enumerable)
        {
            return enumerable != null && enumerable.Any();
        }
    }
}
```

```csharp
namespace Mynamespace.Tests.Extensions
{
    using System.Collections.Generic;
    using System.Linq;
    using Xunit;

    public class IEnumerableExtensionsTests
    {
        [Fact]
        public void SafeAny_MultipleElements_ReturnsTrue()
        {
            // Arrange
            var integers = GetDummyIEnumerable(numberOfElements: 3);

            // Act
            var result = integers.SafeAny();

            // Assert
            Assert.True(result);
        }

        private IEnumerable<int> GetDummyIEnumerable(int numberOfElements)
        {
            for (int i = 0; i < numberOfElements; i++)
            {
                yield return i;
            }
        }

        // ...
    }
}
```

### Expected behavior
I would expect the same behavior for both classes, at least. However, I believe that classes starting with 'I' from interface shouldn't trigger this warning.

### Actual behavior
IEnumerableExtensionsTests triggers the warning while IEnumerableExtensions doesn't.

### Related information
Visual Studio 2019 v16.4.5
SonarLint for Visual Studio 2019 v4.20.0.16339
.NET Core 3.1
Windows 10

**Comment by pavel-mikula-sonarsource** [comment]
Hello @NelsonPRSousa,

Rule is intended to complain about every class named `ISomething` conflicting with I-prefix for interfaces. But that would also complain with `IEnumerableExtensions`, resp. wherever your `IInterfaceHelpers` extensions are. For this reason, classes marked as `static` are not reported. This explains why rule behaves differently for your classes.

Your `IEnumerableExtensionsTests` class name is legit and rule is causing False-Positive in this case.
